### PR TITLE
[Duplicate of #7100] Integrating MongoDB with java

### LINKED ISSUE #4206: Integrating MongoDB with Java
#### Proposed title of article
How MongoDB integrate with JAVA 

#### Proposed article introduction
Here, we are going to learn how to connect MongoDB with java. MongoDB is based on a document model and can be written in C++, JavaScript and python.

#### Key takeaways 
After reading the article the reader should have learnt the following:
1. What MongoDB is
2. MongoDB vs Relational database structure
3. Required tools for connection
5. Creating MongoDB collection
6. Selecting a created MongoDB collection
7. What a document is and how it works with different anomalies(insert, update and delete)

#### Article quality
My topic involves basic knowledge that is easy to understand.

#### References
N/A

### PR DESCRIPTION
**Pull request by muthoni2-lucy**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #4206 

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by muthoni2-lucy** [comment]
Hey @srishilesh i've created a new pr,,Hoping it wont be closed again.

**Comment by srishilesh** [comment]
> Hey @srishilesh i've created a new pr,,Hoping it wont be closed again.

Hey @muthoni2-lucy ,

If you had addressed all my suggestions at once, there would be no reason to close PRs.

I hope you've addressed all the changes atleast now.

If you still need more time to make this PR revision free/readily publishable, you still have time. I hope you have not missed out any of my suggestions.

Will take a look ASAP.

**Comment by muthoni2-lucy** [comment]
I only have an issue with the image src can you please tell me if the format i've used is right

**Comment by muthoni2-lucy** [comment]
hey @srishilesh  how is the PR?

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+### Table of Contents 
+- [Prerequisites](#prerequisite) 
+- [Relational Database Management Systems vs MongoDB](#Relational-Database-Management-Systems-vs-MongoDB)
+-  [Connecting MongoDB with Java](#Connecting-MongoDB-with-Java) 
+- [Creating MongoDB collection](#Creating-MongoDB-collection)
+- [Insert data into the collection](#Insert-data-into-the-collection) 
+- [Update data in the collection](#Update-data-in-the-collection)
+- [Delete data in the collection](#Delete-data-in-the-collection) 
+- [Delete the collection](#Delete-the-collection)
+- [Eclipse versus Intellij Idea in MongoDB integration with Java](#Eclipse-versus-Intellij-Idea-in-MongoDB-integration-with-java)
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:15]: `(#Relational-Database-Management-Systems-vs-MongoDB)`

In this part, after `#`, all the letters must be in lower case separated  by `-`.

Ensure that the topic name is same as this.

Use VS code markdown extensions to ensure if the hyperlinks is working as expected.

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+- A basic understanding of the Java language
+- Eclipse IDE installed on your OS- 
+- MongoDB installed on your OS
+- MongoDB Java Connector
+- Java JDK 
+
+### Relational Database Management Systems vs MongoDB
+#### MongoDB is a non-relational database, while Relational Database Management System is a relational database.
+
+Relational databases store data in rows and columns, also known as records. A relational database works by the use of keys to connect information from multiple tables, e.g., foreign keys and primary keys. On the other hand, data get stored in a non-relational database. There are no tables, rows, primary keys, or foreign keys, as there are in a relational database. MongoDB makes use of documents, which get stored in the form of BSON documents.
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:29]: `MongoDB makes use of documents, which get stored in the form of BSON documents.`

BSON documents? Explain.

Add any sample examples. And, reference links too

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+MySQL, Oracle, IBM DB2, and Microsoft SQL Server are some of the most popular RDBMS.
+MongoDB, CouchDB, CouchBase, Cassandra, HBase, Redis, Riak, and Neo4J are the popular NoSQL databases.
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:31]: Add links for each DB

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+
+#### MongoDB uses collections, whereas RDBMS uses tables.
+Collections are how MongoDB organizes documents. In relational databases, collections it makes use of tables.
+[Differences between collections and tables.](https://docs.mongodb.com/manual/core/databases-and-collections/)
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:35]: You have not addressed this suggestion.

I had asked you to explain what a collection is.

Comment (srishilesh) [reply content/articles/Integrating-mongodb-with -java/index.md:35]: `In relational databases, collections it makes use of tables.`

Collections makes use of tables? What do you mean? Can you brief about this?

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+
+
+#### No schema definition is required in MongoDB, but a clear schema is required in RDBMS.
+MongoDB allows unstructured data to be stored that is building your application without first defining the schema. Before adding data to a relational database, you must first define your schema and this affects cchanging the schema structure of a relational database as it'll be expensive and time-consuming. A schema is the structure of the database that is described in a formal language supported by the database management system. Since MongoDB does not require a predefined schema, it is much easier to update as data and requirements are changing.
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:39]: This paragraph needs rephrasing.

You are beating around the bush. Please statement your point directly, and explain in simple english.

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+#### Scaling in MongoDB and Relational DatabasesThe database must be scaled.
+In MongoDB, it's easy to scale, while in RDBMS, the options for scalability are more limited.
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:42]: Again, you have not addressed my suggestion here.

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+
+### Connecting MongoDB with Java
+
+To connect MongoDB with Java, you'll need to do the following.
+In Eclipse, create a new project. Right-click on the file created, click on properties, and on the Java build path, you'll add the MongoDB Java connector that you downloaded and then click on AddExternalJARS to create a connection between MongoDB and Java. Make sure the MongoDB server is running before using Eclipse or you can follow this link [if you're having any challenges](https://www.mongodb.com/java#:~:text=To%20connect%3A,mongodb%3A%2F%2Flocalhost%3A27017'.)
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:47]: `you'll add the MongoDB Java connector that you downloaded`

Where would you find this?

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+
+#### How to run the MongoDB server
+- Open two command prompt windows
+- The installed MongoDB in local drive c is navigated using one of the cmd prompts, ```bash```. If you haven't installed your MongoDB, you can get help from this link. [Installing MongoDB](https://docs.mongodb.com/guides/server/install/)
+```
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:52]:
```suggestion
```bash

```

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+- After creating the folder, type mongod, and after running, it starts the MongoDB server. The current cmd prompt should not be closed.
+``` 
+C:\Program Files\MongoDB\Server\5.0\bin>mongod
+```
+- To run commands on MongoDB, you need to run the mongo command in the other cmd prompt and type the command for MongoDB to open the shell.
+```
+ C:\ Program Files\MongoDB\Server\5.0\bin>mongo
+```
+After you run the code, you can show how many databases are currently available within your MongoDB.
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:70]: What is `mongdb` and `mongo`?

I don't really understand the difference.

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+
+![Available Databases](/engineering-education/integrating-mongodb-with-java/available-databases.png).
+
+#### Connectivity
+In this tutorial, you'll create a database called SchoolManagementSystems and populate it with a collection called Students. The schoolManagementSystem in simpler terms is the database name while the students will be the table name. But since in MongoDb there is no use of tables youll use collections and it'll be named students. 
+
+Every collection must contain data, and in this tutorial, I'll include the names of two students.
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:77]: Did you not see my reply to your comment in your previous PR?

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+import com.mongodb.client.MongoCollection; import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+
+public static void main(String[] args) throws UnknownHostException {MongoClient mongoClient= new MongoClient( "localhost", 27017);//connecting to the database
+MongoDatabase db=mongoClient.getDatabase("SchoolManagementSystem");
+
+MongoCollection> collection<Document>collection=db.getCollection("students");
+}
+
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:130]: Format the code neatly

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+### Insert data into the collection
+
+Add the following code to your Java class after the created collection.
+```java
+//inserting a new document in the collection which will be student number one.
+Document doc=new Document("name","Christine Muthoni");doc.append("id", "CT201/100091/20");doc.append("course", "Clinical Medicine");doc.append("Gender", "Female");collection.insertOne(doc);System.out.println("Insert is completed");
+
+//insert a new document in the collection, which will be student number two. 
+Document doc2=new Document("name", "Esther Wacera");doc2.append("id", "CT201/10999/14");doc2.append("course", "Civil Engineering");doc2.append("Gender", "Female");collection.insertOne(doc2);System.out.println("Insert is completed");
+```
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:146]: Format them neatly

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 13]
```diff
@@ -0,0 +1,241 @@
...
+In the cmd terminal type
+``` 
+myCursor=db.students.find({})
+```
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:150]: Are you sure you type this in the terminal?

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+}
+```
+In the main class, you'll start the mongo client using the following code snippet.
+  
+```java
+mongoClient= new MongoClient(new MongoClientURL("mongodb://localhost:27017"));
+```
+To instantiate the database add the following code.
+```java
+database= mongoClient.getDB(dbname: "SchoolManagementSystem);
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:228]: Syntax error

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 15]
```diff
@@ -0,0 +1,241 @@
...
+```
+To create a collection you use
+```java
+collectiontest=database.getCollection(name:"student");
+```
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:233]: What are you doing here? Explain

File: content/articles/Integrating-mongodb-with -java/index.md [hunk 16]
```diff
@@ -0,0 +1,241 @@
...
+
+[More on connecting MongoDB with Java using ecllipse](https://www.jetbrains.com/help/idea/mongodb.html)
+
+### Conclusion
+Java and MongoDB are two of the most popular software development technologies, so it's no surprise that they complement each other well. Whether you're developing on a local or on-premises MongoDB installation or going cloud-first with MongoDB Atlas, the MongoDB Java Driver and surrounding ecosystem make integrating MongoDB into your development process a breeze. In this tutorial, we have looked at setting up a MongoDB server. We have also looked at how to establish a connection between MongoDB and Java and how to operate with different CRUD operations. I hope you found this article helpful.!
+ [This Github URL will be of help](https://github.com/dboissier/mongo4idea)
```
Comment (srishilesh) [inline content/articles/Integrating-mongodb-with -java/index.md:239]: This is not your project.

The codes that you used in the tutorial must be here.

**Review by srishilesh — CHANGES_REQUESTED** [review]
Hey @muthoni2-lucy ,

I don't have a choice other than to close this PR.

The state of the PR looks very same from your previous one's. You have not carefully addressed all the suggestions. I find too many typos, grammatical check was not done, code formatting is not done, and there are some discrepancies in markdown formatting.

If all the requested changes are not done by the next PR, we will be closing this topic.

Ping me via Slack, if any questions.

Thanks.

**Comment by srishilesh** [comment]
@muthoni2-lucy 
We do not provide revision services here. If I were to make the changes, I would have to spend too much time on this.

That's the reason I have closed this.

File: content/articles/Integrating-mongodb-with -java/index.md (hunk 13 above)
Comment (muthoni2-lucy) [reply content/articles/Integrating-mongodb-with -java/index.md:150]: hey @srishilesh  
Yes i'm sure,,
